CONTRIBUTING.md

```diff
@@ -40,6 +40,138 @@ GitHub provides additional document on [forking a repository](https://help.githu
 [creating a pull request](https://help.github.com/articles/creating-a-pull-request/).
 
 
+### Running Tests in a local Environment (Docker Compose)
+
+#### Unit Tests
+
+To verify unit tests pass run:
+
+```bash
+mvn clean test
+```
+
+#### Formatting
+
+The CI pipeline will verify that source code files are correctly formatted (using `com.spotify.fmt:fmt-maven-plugin:check`) and license headers are present (using )
+
+To verify correct formatting of your code run:
+
+```bash
+mvn com.spotify.fmt:fmt-maven-plugin:check
+mvn com.mycila:license-maven-plugin:check
+```
+
+> **Note**  
+> When you run `mvn package` the formatter will auto-format files.
+
+#### Packaging
+
+To create the (shaded) JAR(s) run:
+
+```bash
+mvn clean package -Drevision=$(git describe --tags --always)
+```
+
+#### Integration Tests with Docker Compose
+
+To execute the integration tests (requires [Docker](https://docker.com/)) run:
+
+```bash
+export KAFKA_VERSION=3.5.0
+export COMPOSE_FILE=e2e/docker_compose.yaml
+mvn clean verify -Drevision=$(git describe --tags --always)
+```
+
+#### Running a local Kafka Connect Test Environment
+
+Requirements:
+
+- Maven
+- Docker and `docker-compose` CLI
+- `curl`
+- A way to create AWS resources (unless LocalStack is used), e.g. using `aws` CLI
+
+These steps assume you are familiar with EventBridge Event Buses, Rules, and Targets. Consult the EventBridge
+[documentation](https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-what-is.html) for instructions how to set up
+those resources. 
+
+If [Docker](https://docker.com/) and the `docker-compose` CLI is installed, a Kafka test environment can be created
+locally e.g., to create the local environment:
+
+```bash
+# create the JAR which will be loaded into Kafka Connect (Docker Volume)
+mvn clean package -Drevision=$(git describe --tags --always)
+
+# move to the e2e directory
+cd e2e
+
+# set credentials used by the connector to send events to EventBridge (or assume a corresponding role)
```

Comment:
This is only necessary if credentials aren't set already through other means, right? If so, we should recommend skipping when unnecessary.